arch/riscv/dts/sun20i-d1-nezha.dts

```diff
@@ -270,3 +279,238 @@
 	usb1_vbus-supply = <&reg_vcc>;
 	status = "okay";
 };
+
+&pio {
+	gpio-line-names =
+		/* Port A */
+		"PA0",
+		"PA1",
+		"PA2",
+		"PA3",
+		"PA4",
+		"PA5",
+		"PA6",
+		"PA7",
+		"PA8",
+		"PA9",
+		"PA10",
+		"PA11",
+		"PA12",
+		"PA13",
+		"PA14",
+		"PA15",
+		"PA16",
+		"PA17",
+		"PA18",
+		"PA19",
+		"PA20",
+		"PA21",
+		"PA22",
+		"PA23",
+		"PA24",
+		"PA25",
+		"PA26",
+		"PA27",
+		"PA28",
+		"PA29",
+		"PA30",
+		"PA31",
+		/* Port B */
+		"PB0 [gpio2/twi2-sck]",
+		"PB1 [gpio1/twi2-sda]",
+		"PB2",
+		"PB3 [gpio24-i2s2-din]",
+		"PB4 [gpio25-i2s2-dout]",
+		"PB5 [gpio7/i2s-clk]",
+		"PB6 [gpio22/i2s2-lrck]",
+		"PB7",
+		"PB8 [gpio4/uart0-txd]",
+		"PB9 [gpio4/uart0-rxd]",
+		"PB10",
+		"PB11",
+		"PB12 [gpio9]",
+		"PB13",
+		"PB14",
+		"PB15",
+		"PB16",
+		"PB17",
+		"PB18",
+		"PB19",
+		"PB20",
+		"PB21",
+		"PB22",
+		"PB23",
+		"PB24",
+		"PB25",
+		"PB26",
+		"PB27",
+		"PB28",
+		"PB29",
+		"PB30",
+		"PB31",
+		/* Port C */
+		"PC0",
+		"PC1 [gpio21]",
+		"PC2",
+		"PC3",
+		"PC4",
+		"PC5",
+		"PC6",
+		"PC7",
+		"PC8",
+		"PC9",
+		"PC10",
+		"PC11",
+		"PC12",
+		"PC13",
+		"PC14",
+		"PC15",
+		"PC16",
+		"PC17",
+		"PC18",
+		"PC19",
+		"PC20",
+		"PC21",
+		"PC22",
+		"PC23",
+		"PC24",
+		"PC25",
+		"PC26",
+		"PC27",
+		"PC28",
+		"PC29",
+		"PC30",
+		"PC31",
+		/* Port D */
+		"PD0",
+		"PD1",
+		"PD2",
+		"PD3",
+		"PD4",
+		"PD5",
+		"PD6",
+		"PD7",
+		"PD8",
+		"PD9",
+		"PD10 [gpio16/spi1-ce0]",
+		"PD11 [gpio15/spi1-clk]",
+		"PD12 [gpio12/spi1-mosi]",
+		"PD13 [gpio13/spi1-miso]",
+		"PD14 [gpio18/spi1-hold]",
+		"PD15 [gpio20/spi1-wp]",
+		"PD16",
+		"PD17",
+		"PD18",
+		"PD19",
+		"PD20",
+		"PD21",
+		"PD22 [gpio3/pwm]",
+		"PD23",
+		"PD24",
+		"PD25",
+		"PD26",
+		"PD27",
+		"PD28",
+		"PD29",
+		"PD30",
+		"PD31",
+		/* Port E */
+		"PE0",
+		"PE1",
+		"PE2",
+		"PE3",
+		"PE4",
+		"PE5",
+		"PE6",
+		"PE7",
+		"PE8",
+		"PE9",
+		"PE10",
+		"PE11",
+		"PE12",
+		"PE13",
+		"PE14",
+		"PE15",
+		"PE16",
+		"PE17",
+		"PE18",
+		"PE19",
+		"PE20",
+		"PE21",
+		"PE22",
+		"PE23",
+		"PE24",
+		"PE25",
+		"PE26",
+		"PE27",
+		"PE28",
+		"PE29",
+		"PE30",
+		"PE31",
+		/* Port F */
+		"PF0",
+		"PF1",
+		"PF2",
+		"PF3",
+		"PF4",
+		"PF5",
+		"PF6",
+		"PF7",
+		"PF8",
+		"PF9",
+		"PF10",
+		"PF11",
+		"PF12",
+		"PF13",
+		"PF14",
+		"PF15",
+		"PF16",
+		"PF17",
+		"PF18",
+		"PF19",
+		"PF20",
+		"PF21",
+		"PF22",
+		"PF23",
+		"PF24",
+		"PF25",
+		"PF26",
+		"PF27",
+		"PF28",
+		"PF29",
+		"PF30",
+		"PF31",
```

Comment:
While there is space reserved for 32 pins in each port, most ports have many fewer pins. For example, port F only has pins 0-6. So the gaps should use empty strings. And the [binding documentation](https://www.kernel.org/doc/Documentation/devicetree/bindings/gpio/gpio.txt) suggests only defining the pins that are actually used or exposed to the user (emphasis mine):

> Optionally, a GPIO controller may have a "gpio-line-names" property. This is
an array of strings defining the names of the GPIO lines going out of the
GPIO controller. This name should be the most meaningful producer name
for the system, such as a rail name indicating the usage. *Package names
such as pin name are discouraged*: such lines have opaque names (since they
are by definition generic purpose) and such names are usually not very
helpful. For example "MMC-CD", "Red LED Vdd" and "ethernet reset" are
reasonable line names as they describe what the line is used for. "GPIO0"
is not a good name to give to a GPIO line. *Placeholders are discouraged:
rather use the "" (blank string) if the use of the GPIO line is undefined
in your design*. The names are assigned starting from line offset 0 from
left to right from the passed array. An incomplete array (where the number
of passed named are less than ngpios) will still be used up until the last
provided valid line index.